memor/prompt.py

```diff
@@ -331,6 +332,13 @@ def to_dict(self, save_template: bool = True) -> Dict[str, Any]:
             del data["template"]
         return data
 
+    def regenerate_id(self) -> None:
+        """Regenerate ID."""
+        new_id = self._id
+        while new_id == self.id:
+            new_id = generate_message_id()
+        self._id = new_id
```

Comment:
Why it's not just:
```python
self._id = generate_message_id()
```
?
Because it's as likely for it to generate repetitive id here as it is for the new object.

Reply:
You are right. However, when we have a previous UUID, it's better to double-check before generating a new one. In the beginning, we do not have access to the previous UUID.

Reply:
I disagree, we could have a list of all generated uids and check from that. I like to go with all or nothing (either check every instances, or check none).